src/cryptography/x509.py

```diff
@@ -141,6 +142,61 @@ def __len__(self):
         return len(self._attributes)
 
 
+OID_BASIC_CONSTRAINTS = ObjectIdentifier("2.5.29.19")
+
+
+class Extension(object):
+    def __init__(self, oid, critical, value):
+        if not isinstance(oid, ObjectIdentifier):
+            raise TypeError(
+                "oid argument must be an ObjectIdentifier instance."
+            )
+
+        if not isinstance(critical, bool):
+            raise TypeError("critical must be a boolean value")
+
+        self._oid = oid
+        self._critical = critical
+        self._value = value
+
+    oid = utils.read_only_property("_oid")
+    critical = utils.read_only_property("_critical")
+    value = utils.read_only_property("_value")
+
+    def __repr__(self):
+        return ("<Extension(oid={oid}, "
+                "critical={critical}, value={value})>").format(
+            oid=self.oid, critical=self.critical, value=self.value
+        )
```

Comment:
This can be written slightly more succinctly as `"<Extension(oid={0.oid}, critical={0.critical}, value={0.value})>".format(self)`